server/lib/src/entry.rs

```diff
@@ -2353,6 +2372,37 @@ impl Entry<EntryReduced, EntryCommitted> {
     }
 }
 
+fn resolve_interim(
+    scim_value_intermediate: ScimValueIntermediate,
+    read_txn: &mut QueryServerReadTransaction,
+) -> Result<Option<ScimValueKanidm>, OperationError> {
+    match scim_value_intermediate {
+        ScimValueIntermediate::Refer(uuid) => {
+            if let Some(option) = read_txn.uuid_to_spn(uuid)? {
+                Ok(Some(ScimValueKanidm::EntryReference(ScimReference {
+                    uuid,
+                    value: option.to_proto_string_clone(),
+                })))
+            } else {
+                // TODO: didn't have spn, fallback to uuid.to_string ?
```

Comment:
Doesn't uuid_to_spn already do this?

Reply:
Doesn't appear like that according to the test
https://github.com/ToxicMushroom/kanidm/blob/957aaabe5e6cb6a950ee9f8aef63d368e7577241/server/lib/src/server/mod.rs#L2362
      VV
 ```rust
      
        // Name doesn't exist
        let r1 = server_txn.uuid_to_spn(uuid!("bae3f507-e6c3-44ba-ad01-f8ff1083534a"));
        // There is nothing.
        assert_eq!(r1, Ok(None));
   ```